### PR TITLE
feat: add methods for whether firmware update is in progress

### PR DESCRIPTION
**Pull request by raman325**
There's no reason why the driver can't return whether a firmware update is in progress. Instead of relying on the application to track it, consider cases like zwavejs2mqtt where there are two applications (z2m and zwave-js-server) that can initiate updates. IMO it needs to be centralized in the driver

### CODE REVIEW

File: packages/zwave-js/src/lib/controller/Controller.ts [hunk 1]
```diff
@@ -534,6 +534,13 @@ export class ZWaveController extends TypedEventEmitter<ControllerEventCallbacks>
 		this.driver.cacheSet(cacheKeys.controller.supportsSoftReset, value);
 	}
 
+	public get isAnyFirmwareUpdateInProgress(): boolean {
```
Comment (AlCalzone) [inline packages/zwave-js/src/lib/controller/Controller.ts]: Should we call this `isOTAFirmwareUpdateInProgress` in order to distinguish node updates from the OTW update for the controller (which I'll eventually add)?

Also should this be a getter/property or a method? Feels more like a method at first glance.

File: packages/zwave-js/src/lib/node/Node.ts [hunk 2]
```diff
@@ -795,6 +795,10 @@ export class ZWaveNode
 		}
 	}
 
+	public get isFirmwareUpdateInProgress(): boolean {
```
Comment (AlCalzone) [inline packages/zwave-js/src/lib/node/Node.ts]: Should this be a getter/property or a method? Feels more like a method at first glance.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Thanks - two/three thoughts:

File: packages/zwave-js/src/lib/controller/Controller.ts (hunk 1 above)
Comment (raman325) [reply packages/zwave-js/src/lib/controller/Controller.ts]: good idea, and we can make it a method

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**